File: client/src/boundary/model/BoundaryGame.ts
import BoundaryBoard from "./BoundaryBoard";

export default class BoundaryGame {
  readonly id: string;
  readonly state: string;
  readonly board: BoundaryBoard;

  constructor(id: string, state: string, board: BoundaryBoard) {
    this.id = id;
    this.state = state;
    this.board = board;
  }
}

File: server/battleships/use_cases/api/src/main/java/eu/optas/use_cases/api/BoundaryGame.java
package eu.optas.use_cases.api;

import eu.optas.utils.GameState;

public class BoundaryGame {
    private final String id;
    private final GameState state;
    private final BoundaryBoard board;
    private final int hitsRemaining;
    private final int shipsDestroyed;

    public BoundaryGame(String id, GameState state, BoundaryBoard board, int hitsRemaining, int shipsDestroyed) {
        this.id = id;
        this.state = state;
        this.board = board;
        this.hitsRemaining = hitsRemaining;
        this.shipsDestroyed = shipsDestroyed;
    }

    public String getId() {
        return id;
    }

    public GameState getState() {
        return state;
    }

    public BoundaryBoard getBoard() {
        return board;
    }

    public int getHitsRemaining() {
        return hitsRemaining;
    }

    public int getShipsDestroyed() {
        return shipsDestroyed;
    }
}
